analyzer/src/funTest/assets/projects/synthetic/gomod-expected-output.yml

```diff
@@ -20,42 +20,13 @@ project:
     dependencies:
     - id: "GoMod::github.com/fatih/color:v1.7.0"
       linkage: "PROJECT_STATIC"
-    - id: "GoMod::github.com/hashicorp/go-multierror:v1.0.0"
```

Comment:
Commit message nit: I believe there should be no comma in "contain packages, which are".

Reply:
Out of curiosity: Can you explain to me what "not directly used by the main module" means in more detail? I mean, probably no transitive dependency is directory used (also) by the main module, but still some transitive dependencies remain after your change (which is good). Is it some sort of "dead code elimination" that's going on here, so if there is no code path that really makes use of a transitive dependency, it gets removed?